packages/autocomplete-plugin-recent-searches/src/usecases/localStorage/createLocalStorage.ts

```diff
@@ -0,0 +1,27 @@
+import { isLocalStorageSupported } from './isLocalStorageSupported';
+
+type LocalStorageProps = {
+  key: string;
+};
+
+export function createLocalStorage<TItem>({ key }: LocalStorageProps) {
+  if (!isLocalStorageSupported()) {
+    return {
+      setItem() {},
+      getItem() {
+        return [];
+      },
+    };
+  }
+
+  return {
+    setItem(items: TItem[]) {
+      return window.localStorage.setItem(key, JSON.stringify(items));
+    },
+    getItem(): TItem[] {
```

Comment:
what about `setItems` and `getItems`?

Reply:
I originally matched the `localStorage` signature but those might make more sense now that they're abstractions indeed.

Reply:
Hmm actually we're referring solely to the local storage here so that's debatable.